devtools/run_conda_lock.py

```diff
@@ -91,6 +96,28 @@ def per_platform_env(py_ver: str, extras=[], dev=False, suffix="") -> None:
             f.write(env_file_variables_section_)
 
 
+def patch_absolute_path(file: Path) -> None:
+    """
+    Patch the given file to remove reference with absolute file path.
+    """
+
+    abs_path_base = str(_environments_folder.absolute().parent) + os.sep
+    assert abs_path_base.endswith(os.sep)
```

Comment:
This is redundant since you just added it to the string.  Also, os.path has some convenience functions for working with path strings.  I use os.path.join a lot which will join strings with the os.sep by default and will only use one separator in case one of the strings already has one.  Might be useful here.